grammars/erlang.bnf

```diff
@@ -321,6 +326,50 @@ argument_definition_list ::= '(' argument_definition? (',' argument_definition)*
 clause_guard ::= when <<guarded guard>> {pin=1}
 clause_body ::= '->' exprs {pin=1}
 
+app_expression ::= '{' app_application_keyword ',' app_atom_expression ',' app_parameters '}' {elementType=tuple_expression}
+app_application_keyword ::= &'application' q_atom {elementType=config_expression}
+app_atom_expression ::= q_atom {elementType=config_expression}
+app_parameters ::= '[' app_parameter? (',' app_parameter)* ']' {elementType=list_expression}
+
+app_parameter ::= '{'
+   (app_description_keyword ',' string_literal
+  | app_id_keyword ',' string_literal
+  | app_vsn_keyword ',' string_literal
+  | app_modules_keyword ',' app_modules
+  | app_maxP_keyword ',' integer
+  | app_maxT_keyword ',' integer
+  | app_registered_keyword ',' app_modules
+  | app_included_applications_keyword ',' app_applications
+  | app_applications_keyword ',' app_applications
+  | app_env_keyword ',' app_env
+  | app_mod_keyword ',' app_mod
+  | app_start_phases_keyword ',' expression
+  | app_runtime_dependencies_keyword ',' app_runtime_deps
+  | config_exprs)
+  '}' {elementType=tuple_expression}
```

Comment:
It's still somewhat incomplete. Some of your terms are wrapped in a ConfigExpression, others - aren't. Your PSI should be uniform, so that it can be traversed the same way for all cases.